setup.py

```diff
@@ -164,35 +168,50 @@ def wrapper(*args, **kw):  # type: ignore
     return wrapper
 
 
+_compute_shuffled_index = compute_shuffled_index
+compute_shuffled_index = cache_this(
+    lambda index, index_count, seed: (index, index_count, seed),
+    _compute_shuffled_index, lru_size=SLOTS_PER_EPOCH * 3)
+
+_get_total_active_balance = get_total_active_balance
+get_total_active_balance = cache_this(
+    lambda state: (state.validators.hash_tree_root(), state.slot),
```

Comment:
This can cache based on current epoch instead of current slot.

Also this cache is only safe as long as lookups aren't for more than `MAX_SEED_LOOKAHEAD` epochs in the future

Reply:
> This can cache based on current epoch instead of current slot.

It's only called during epoch transition, so effectively the same. I will change it to epoch key though.

> Also this cache is only safe as long as lookups aren't for more than MAX_SEED_LOOKAHEAD epochs in the future

The cache is safe the function is not. I am not changing function behavior with the memoization

Reply:
ah, right. the function won't allow the call.